scripts/FigS1B-D_method-metrics-comparison.R

```diff
@@ -0,0 +1,223 @@
+
+# add per cell and per feature info to each sce 
+all_sces <- list(af_sces, cellranger_sces) |> 
+  purrr::set_names(c("Alevin-fry", "Cell Ranger")) |> 
+  purrr::map(\(list){
+    list |> 
+      purrr::map(\(sce) {
+        sce |> 
+          scuttle::addPerCellQCMetrics(subsets = list(mito = mito_genes[mito_genes %in% rownames(sce)])) |>
+          scuttle::addPerFeatureQCMetrics()
+      })
+  })
+
+# Prep for plotting ------------------------------------------------------------
+
+# read in library metadata
+library_df <- readr::read_tsv(library_metadata_file) |> 
+  dplyr::select(run_id = scpca_run_id, scpca_library_id)
+
+# read in suspension palette 
+suspension_palette <- readr::read_tsv(suspension_palette_file) |> 
+  dplyr::rename(seq_unit = suspension_type)
+
+# create a data frame with coldata info for each tool, run id combo 
+coldata_df <- all_sces |> 
+  purrr::map_df(~ purrr::map_df(.x, scpcaTools::coldata_to_df, .id = "run_id"),
+    .id = "tool"
+    ) |> 
+  # create new columns with seq unit 
+  dplyr::mutate(seq_unit = dplyr::case_when(run_id %in% single_cell ~ "Single-cell",
+                                            run_id %in% single_nuclei ~ "Single-nuclei")) |>
+  dplyr::left_join(library_df) |>
+  dplyr::mutate(plot_id = glue::glue("{scpca_library_id} ({seq_unit})")) |> 
+  dplyr::left_join(suspension_palette)
+
+# filter for cells that are found in both af + cellranger
+cell_counts <- coldata_df |>  
+  dplyr::count(cell_id, run_id)
+
+common_cells <- cell_counts |>
+  dplyr::filter(n == 2) |>
+  dplyr::pull(cell_id)
+
+coldata_common <- coldata_df |>
+  dplyr::filter(cell_id %in% common_cells)
+
+# Set up colors ----------------------------------------------------------------
+
+# read in method palette colors 
+method_palette <- readr::read_tsv(method_palette_file)
+
+# get list of all colors 
+method_colors <- method_palette$color |> 
+  purrr::set_names(method_palette$method)
+
+# suspension colors 
+suspension_palette <- readr::read_tsv(suspension_palette_file)
+
+# get list of all colors 
+suspension_colors <- suspension_palette$color |> 
+  purrr::set_names(suspension_palette$suspension_type)
+
+# define suspension backgrounds to use on facet strips 
+backgrounds <- list(element_rect(fill = suspension_colors[["Single-cell"]]),
+                    element_rect(fill = suspension_colors[["Single-cell"]]),
+                    element_rect(fill = suspension_colors[["Single-cell"]]),
+                    element_rect(fill = suspension_colors[["Single-nuclei"]]),
+                    element_rect(fill = suspension_colors[["Single-nuclei"]]),
+                    element_rect(fill = suspension_colors[["Single-nuclei"]]))
+
+# Plot cell metrics ------------------------------------------------------------
+
+# create combined UMI per cell plot 
+umi_plot <- ggplot(coldata_common, aes(x = sum, color = tool)) + 
+  geom_density() + 
+  ggh4x::facet_wrap2(~plot_id,
+                     scales = "free",
```

Comment:
I think we don't want "free" scales for these plots? I think the scales are close enough to each other to plot everything on the same sets of axes, unless it looks terrible.